pkg/apis/keda/v1alpha1/types.go

```diff
@@ -18,7 +25,9 @@ type ScaledObject struct {
 
 // ScaledObjectSpec is the spec for a ScaledObject resource
 type ScaledObjectSpec struct {
+	ScaleType       string          `json:"scaleType"`
```

Comment:
@stgricci wondering if this should be treated as a "enum". [Example](https://github.com/coreos/etcd-operator/blob/master/pkg/apis/etcd/v1beta2/backup_types.go)

type ScaledObjectScaleType string

const (
	ScaleTypeDeployment ScaledObjectScaleType = "deployment"
	ScaleTypeJob                                                      = "job"
)

type ScaledObjectSpec struct {
  ScaleType     ScaledObjectScaleType   `json:"scaleType"`
}


```